### DISCUSSION TITLE
Check if a machine is 'settled'

### DISCUSSION THREAD

**Discussion by doender** [statelyai/xstate#4766] [Q&A]
In our backend we have a machine that can go through several states based on a single triggering event, e.g. `waiting -> adding info -> ready for submission -> submitting -> submitted. Within the controller we basically only wait for the first transitions to happen (`waiting -> adding info -> ready for submission`) and the other transitions happen after the response has been sent.

My concern is that since we never stop the actor (because `submitted` is *not* a final state and it could take weeks before the final state is hit) this will cause memory to increase. I think the solution would be to check whether the machine is 'settled', that is, there are no actors running and no automatic transitions going to happen through `always` or `after`. In that case we could stop the machine and when a new event comes in, we just create a new machine and rehydrate it. 

Is this the right approach, and is there a way to check whether a machine is settled?

**Comment by tamlyn** [comment]
Yes, I have exactly this problem. In order to know when to serialise a long-running actor in a backend context, I need to distinguish when a machine actor is waiting for an external event vs waiting for a child actor to complete.

**Comment by davidkpiano** [comment]
The transition API that's coming up soon (https://github.com/statelyai/xstate/pull/4954) may help with this, since it allows you to have full control over how your state machines run.
